test/JDBC/input/xml/xml_value-vu-verify.sql

```diff
@@ -0,0 +1,781 @@
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)');
+GO
+
+-- Acceptable argument types
+DECLARE @xml XML = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)');
+GO
+
+-- Unacceptable argument types
+DECLARE @xml XML = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @xml.value(cast('(/artists/artist/@name)[1]' as VARCHAR(100)), 'varchar(100)');
+GO
+
+DECLARE @xml XML = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @xml.value(cast('(/artists/artist/@name)[1]' as text), 'varchar(100)');
+GO
+
+DECLARE @xml XML = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @xml.value(@xml, 'varchar(100)');
+GO
+
+DECLARE @xml VARCHAR(100) = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)');
+GO
+
+DECLARE @xml NVARCHAR(100) = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)');
+GO
+
+DECLARE @xml INT = 1
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)');
+GO
+
+DECLARE @xml BIGINT = 1
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)')
+GO
+
+DECLARE @xml FLOAT = 1
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)')
+GO
+
+DECLARE @xml DECIMAL = 1
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)')
+GO
+
+DECLARE @xml NUMERIC = 1
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)')
+GO
+
+DECLARE @xml BIT = 1
+SELECT @xml.value('(/artists/artist/@name)[1]', 'varchar(100)')
+GO
+
+SELECT XmlColumn.value('(/artists/artist/@name)[1]', 'varchar(100)') FROM babel_5223_xml_value_text
+GO
+
+-- UDT on type VARCHAR
+SELECT VarUDTColumn.value('(/artists/artist/@name)[1]', 'varchar(100)') FROM babel_5223_xml_value_udt
+GO
+
+-- UDT on type IMAGE
+SELECT ImageUDTColumn.value('(/artists/artist/@name)[1]', 'varchar(100)') FROM babel_5223_xml_value_udt
+GO
+
+-- UDT on type XML
+SELECT XmlUDTColumn.value('(/artists/artist/@name)[1]', 'varchar(100)') FROM babel_5223_xml_value_udt
+GO
+
+-- Tests on different return types
+DECLARE @xml XML = '<artists> <artist id="1"/> </artists>'
+SELECT @xml.value('(/artists/artist/@id)[1]', 'varchar(10)');
+GO
+
+DECLARE @xml XML = '<artists> <artist id="1"/> </artists>'
+SELECT @xml.value('(/artists/artist/@id)[1]', 'nvarchar(10)');
+GO
+
+DECLARE @xml XML = '<artists> <artist id="1"/> </artists>'
+SELECT @xml.value('(/artists/artist/@id)[1]', 'char(10)');
+GO
+
+DECLARE @xml XML = '<artists> <artist id="1"/> </artists>'
+SELECT @xml.value('(/artists/artist/@id)[1]', 'nchar(10)');
+GO
```

Comment:
Please add test with datalength() on the output to validate the length of the output.

Reply:
Added